2.preprocess_features/scripts/1.norm_fs_pycytominer.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Perform data preprocessing with pycytominer on each compartment file
+
+# ## Import libraries
+
+# In[1]:
+
+
+import pathlib
+
+import pandas as pd
+from pycytominer import normalize, feature_select
+from pycytominer.cyto_utils import infer_cp_features
+
+
+# ## Set paths and variables
+
+# In[2]:
+
+
+# operations to perform for feature selection
+feature_select_ops = [
+    "variance_threshold",
+    "correlation_threshold",
+    "blocklist",
+    "drop_na_columns"
+]
+
+# Output directory for processed profiles
+output_dir = pathlib.Path("./profiles/processed_profiles")
+output_dir.mkdir(parents=True, exist_ok=True)
+
+# Path to CytoTable converted profiles for each object
+converted_profiles_dir = pathlib.Path("./profiles/converted_profiles/")
+
+# List all the converted files in the directory
+converted_files = [file for file in converted_profiles_dir.iterdir() if file.is_file()]
+
+# Print the converted files that will be processed
+for file in converted_files:
+    print(file)
+
+
+# ## Perform preprocessing on single cell features
+
+# In[3]:
+
+
+for file in converted_files:
+    print(f"Performing pycytominer pipeline for {file.stem}")
+    output_normalized_file = str(
+        pathlib.Path(f"{output_dir}/{file.stem}_normalized.parquet")
+    )
+    output_feature_select_file = str(
+        pathlib.Path(f"{output_dir}/{file.stem}_feature_selected.parquet")
+    )
+
+    # Load in file to process
+    df = pd.read_parquet(file)
+
+    # Set the compartment name to find in the data frame (must be capitalized)
+    compartment_name = file.stem.split("_")[1].capitalize()
```

Comment:
This approach can be susceptible to bugs. For instance, if the column names are: `[NotFeature_data1, NotFeature_data2, Compartment_feature1, Compartment2, feature1]`, your code will return `[NotFeature, Compartment, Compartment2]` therefore mistakenly identify `NotFeature`  as valid feature names. There could be a possibility of retaining metadata features that are incorrectly treated as morphological features.

Reply:
If that's not the case, feel free to ignore this comment c:

Reply:
In this case, each df has only that compartments data without any metadata so I don't think this will entirely be an issue for this context, but thanks for bringing it up. I will keep it in mind!